src/ios-scroll.js

```diff
@@ -0,0 +1,513 @@
+     * scrollTop and our synthetic scroll offset.
+     * @private {number}
+     */
+    this.min_ = -scrollTop;
+
+    /**
+     * The maximum value that we may scroll the scroller to.
+     * @private {number}
+     */
+    this.max_ = Math.round(scrollHeight - this.win_./*OK*/innerHeight);
+
+    /**
+     * The current scroll offset from the body's own scrollTop. After synthetic
+     * scrolling, this value will be added to the scrollTop.
+     * @private {number}
+     */
+    this.offset_ = 0;
+
+    /**
+     * Records the y position of the last touch event, and is used to
+     * determine how much movement happened in the current touch event.
+     * @private {number}
+     */
+    this.reference_ = 0;
+
+    /**
+     * Records the timestamp of the last touch event, and is used to
+     * determine how much time elapsed between now.
+     * @private {number}
+     */
+    this.timestamp_ = 0;
+
+    /**
+     * A calculated velocity that the current scroll gives the doc.
+     * @private {number}
+     */
+    this.velocity_ = 0;
+
+    /**
+     * A total distance between the current offset and where the doc should
+     * end up after accounting for the velocity.
+     * @private {number}
+     */
+    this.amplitude_ = 0;
+
+    /**
+     * The target offset that scroll velocity carries us to.
+     * @private {number}
+     */
+    this.target_ = 0;
+
+    /**
+     * The animation frame id for our synthetic momentum scrolling step
+     * function.
+     * @private {number}
+     */
+    this.autoScrolling_ = 0;
+
+    /**
+     * Used to determine if the user has native scrolled (due to an iframe).
+     * To account for our own scrolling of the body element (synchronizing the
+     * scrollTop and our synthetic scrolling), this value may be null meaning
+     * that this scroll event is expected. If so, it will be set to false after
+     * the expected scroll. Otherwise, it will be true indicating a native
+     * scroll.
+     * @private {boolean|null}
+     */
+    this.bodyScrolled_ = false;
+
+    /**
+     * The synthetic momentum scroll, bound to this context.
+     * @private
+     */
+    this.boundAutoScroll_ = this.autoScroll_.bind(this);
+
+    // We listen on the document level so that, even if the HTML or BODY has
+    // padding, margin, etc, the touch event will still cause scrolling.
+    doc.addEventListener('touchstart', this.touchstart_.bind(this));
+    doc.addEventListener('touchmove', this.touchmove_.bind(this));
+    doc.addEventListener('touchend', this.touchend_.bind(this));
+    // Native scrolling does some weird things. When the user native scrolls
+    // due to an iframe, the only event fired in this doc is a `scroll` on
+    // the body element.
+    body.addEventListener('scroll', this.bodyScroll_.bind(this));
+
+    // Promote the scroller to it's own GPU rendering layer.
+    this.scrollStyle_.willChange = 'transform';
+  }
```

Comment:
Unfortunately not, since the element can be anything, and we don't want this on non-iOS.